library/core/src/array/mod.rs

```diff
@@ -435,4 +436,75 @@ impl<T, const N: usize> [T; N] {
     pub fn as_mut_slice(&mut self) -> &mut [T] {
         self
     }
+
+    /// Borrows each element and returns an array of references with the same
+    /// size as `self`.
```

Comment:
I would say `length` here, since the overall size (as computed by `std::mem::size_of_val`) will likely be different.